src/node/chainstate.h

```diff
@@ -16,19 +18,28 @@ struct Params;
 } // namespace Consensus
 
 namespace node {
-enum class ChainstateLoadingError {
-    ERROR_LOADING_BLOCK_DB,
-    ERROR_BAD_GENESIS_BLOCK,
-    ERROR_PRUNED_NEEDS_REINDEX,
-    ERROR_LOAD_GENESIS_BLOCK_FAILED,
-    ERROR_CHAINSTATE_UPGRADE_FAILED,
-    ERROR_REPLAYBLOCKS_FAILED,
-    ERROR_LOADCHAINTIP_FAILED,
-    ERROR_GENERIC_BLOCKDB_OPEN_FAILED,
-    ERROR_BLOCKS_WITNESS_INSUFFICIENTLY_VALIDATED,
-    SHUTDOWN_PROBED,
+
+struct CacheSizes;
+
+struct InitOptions
+{
+    CTxMemPool* mempool = nullptr;
+    bool block_tree_db_in_memory = false;
+    bool coins_db_in_memory = false;
+    bool reindex = false;
+    bool reindex_chainstate = false;
+    bool prune = false;
+    int64_t check_blocks = DEFAULT_CHECKBLOCKS;
+    int64_t check_level = DEFAULT_CHECKLEVEL;
+    std::function<bool()> check_interrupt;
+    std::function<void()> coins_error_cb;
 };
 
+enum class InitStatus { SUCCESS, FAILURE, FAILURE_INCOMPATIBLE_DB, INTERRUPTED };
```

Comment:
Perhaps we can add some Doxygen comments hinting callers on how they should behave when encountering each case.